python_modules/libraries/dagster-dbt/dagster_dbt/cloud/resources.py

```diff
@@ -181,6 +181,44 @@ def run_job(self, job_id: int, **kwargs) -> Dict[str, Any]:
         )
         return resp
 
+    def get_runs(
+        self,
+        include_related: Optional[List[str]] = [],
+        job_definition_id: Optional[int] = None,
+        order_by: Optional[str] = "-id",
+        offset: int = 0,
+        limit: int = 100,
+    ) -> List[Dict[str, any]]:
+        """
+        Returns a list of runs from dbt cloud. This can be optionally filtered to a specific job
+        using the job_definition_id. It supports pagination using offset and limit as well and
+        can be configured to load a variety of related information about the runs.
+
+        Args:
+            include_related (Optional[List[str]]): A list of resources to include in the response
+                from DBT Cloud. This is technically a required field according to the API, but it
+                can be passed with an empty list where it will only load the default run
+                information. Valid values are "trigger", "job", "repository", and "environment".
+            job_definition_id (Optional[int]): This method can be optionally filtered to only
+                load runs for a specific job id if it is included here. If omitted it will pull
+                runs for every job.
+            order_by (Optional[str]): An identifier designated by DBT Cloud in which to sort the
+                results before returning them. Useful when combined with offset and limit to load
+                runs for a job. Defaults to "-id" where "-" designates reverse order and "id" is
+                the key to filter on.
+            offset (int): An offset to apply when listing runs. Can be used to paginate results
+                when combined with order_by and limit. Defaults to 0.
+            limit (int): Limits the amount of rows returned by the API. Defaults to 100.
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries containing the runs and any included
+                related information.
+        """
+        query_params = f"?include_related={','.join(include_related)}"
```

Comment:
I think this might be better written using https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode

Something like:

```
query_dict = {
    "include_related": include_related or [],
    "order_by": order_by,
    "limit": limit,
}
if job_id:
    query_dict["job_definition_id"] = job_id
return self.make_request("GET", f"{self._account_id}/runs/{urlencode(query_dict)}"
```

This has the benefit of being a bit more readable, and urlencode will automatically escape any non-url friendly characters (although, to be fair that's not all that relevant here)

Reply:
agreed, fwiw I just followed the similar style I saw below. In the past I've typically just let requests handle that as a params arg. But this is much cleaner

Reply:
Ouch you're right -- I definitely have sinned in the past w/ this implementation. I should go back and clean some of this stuff up 😬